### PR TITLE
[ci] Parallelize yarn build

### PR DESCRIPTION
**Pull request by poteto**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #30322
* #30169
* #30072
* #30073
* __->__ #30071
* #30070

### CODE REVIEW

File: scripts/rollup/build-ghaction.js
Comment (poteto) [inline scripts/rollup/build-ghaction.js:1]: I will unfork this later

File: scripts/rollup/build-ghaction.js [hunk 1]
```diff
@@ -0,0 +1,869 @@
+'use strict';
```
Comment (jackpope) [inline scripts/rollup/build-ghaction.js:1]: Review note: diffed this against rollup/build.js locally. Its mostly a copy with some changes to how we get node count from the runner. LGTM, assume we'll remove rollup/build.js once this is stable.

File: .github/workflows/runtime_build.yml [hunk 2]
```diff
@@ -14,6 +14,11 @@ jobs:
   build_and_lint:
     name: yarn build and lint
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # yml is dumb. update the --total arg to yarn build if you change the number of workers
+        worker_id: [0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19]
```
Comment (jackpope) [inline .github/workflows/runtime_build.yml:20]: Curious why 20 workers. Does this match the number of bundles?

**Review by jackpope — APPROVED** [review]

File: .github/workflows/runtime_build.yml (hunk 2 above)
Comment (poteto) [reply .github/workflows/runtime_build.yml:20]: I preserved our current circleci setup, where we have 20 workers for stable and 20 workers for experimental. That's the same as what this config in GH expands to

File: scripts/rollup/build-ghaction.js (hunk 1 above)
Comment (poteto) [reply scripts/rollup/build-ghaction.js:1]: yeah I'm gonna remove the circleci hardcoding later in the stack and unfork these files

**Pull request merged**